controllers/application/controller.go

```diff
@@ -122,6 +123,13 @@ func (r *ApplicationReconciler) Reconcile(ctx context.Context, req reconcile.Req
 		application.Labels = aggregateLabels
 	}
 
+	// Add team label
+	if len(application.Spec.Team) == 0 {
```

Comment:
I think we should add a testcase for an application in a namespace without the team label.